core/block.go

```diff
@@ -39,9 +39,10 @@ type Block struct {
 }
 
 type blockHashMetaInfo struct {
-	First07Block             uint64     // First block that uses the post-0.7.0 block hash algorithm
-	UnverifiableRange        []uint64   // Range of blocks that are not verifiable
-	FallBackSequencerAddress *felt.Felt // The sequencer address to use for blocks that do not have one
+	First07Block             uint64               // First block that uses the post-0.7.0 block hash algorithm
+	UnverifiableRange        []uint64             // Range of blocks that are not verifiable
+	FallBackSequencerAddress *felt.Felt           // The sequencer address to use for blocks that do not have one
+	TxTypesForCommitment     map[Transaction]bool // (Non-invoke) Transaction types to be included in the transactionCommitment
```

Comment:
You don't need to add a `TxTypesForCommitment`. Only `transaction.go` needs to be updated.